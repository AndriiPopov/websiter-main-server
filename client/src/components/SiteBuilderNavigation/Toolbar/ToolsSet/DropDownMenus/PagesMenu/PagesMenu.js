import React from 'react';
import { connect } from 'react-redux';

import * as classes from './PagesMenu.module.css';
import ToolSet from '../../ToolsSet';
import Pages from '../../TopMenuSections/Pages';
import Aux from '../../../../../../hoc/Aux';
import PagesMenuList from './PagesMenuList/PagesMenuList';
import Spinner from '../../../../../UI/Spinner/Spinner';

export const PagesMenu = props => {
    const items = props.pagesStructure.map( page => {
        return {
            ...page,
            name: props.pagesObjects[page.id].name
        }
    });
    return (
        <Aux>
            <header className={classes.Toolbar}>
                <div className={classes.ToolSetsContainer}>
                    <ToolSet isActive >
                        <Pages />
                    </ToolSet>
                </div>
            </header>
            <div className={classes.Content}>
                <PagesMenuList 
                    items={items}
                    currentPage={props.currentPage}
                    pagesObjects={props.pagesObjects}
                    websiteId={props.websiteId}
                    savePagesStructure={props} />
            </div>
            {props.loading ? <Spinner cover /> : null}
        </Aux>
    )
}

const mapStateToProps = state => {
    return {
        currentPage: state.website.currentPage,
        pagesStructure: state.website.pagesStructure,
        pagesObjects: state.website.pagesObjects,
        websiteId: state.website._id,
        loading: state.website.pagesLoading
    }
}

export default connect(mapStateToProps)(PagesMenu);