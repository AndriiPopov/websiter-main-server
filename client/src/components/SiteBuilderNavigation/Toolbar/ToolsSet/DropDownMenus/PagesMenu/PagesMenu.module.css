.Toolbar {
    position: relative;
    top: 0px;
    left: 0px;
    width: 100%;
    z-index: 16777271;
    -webkit-touch-callout: none;
    -webkit-user-select: none;
    -khtml-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
    color: #222;
}

.Toolbar nav {
    width: 100%;
    background: #555;
    height: 30px;
    white-space: nowrap;
    overflow-y: hidden;
    overflow-x: auto;
}

.ToolSetsContainer {
    border-bottom: 1px solid #ccc;
    box-sizing: border-box;
    white-space: nowrap;
    width: 100%;
    background: #eee;
    height: 100px;
    font-size: 12px;
    overflow-x: auto;
    overflow-y: hidden;
}

.Content {
    width: 100%;
    height: 100%;
    min-height: 400px;
    overflow: hidden;
}