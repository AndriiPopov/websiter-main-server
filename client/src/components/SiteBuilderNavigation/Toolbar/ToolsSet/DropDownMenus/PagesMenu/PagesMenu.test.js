import React from 'react';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { PagesMenu } from './PagesMenu';

configure({adapter: new Adapter()});

describe('PagesMenu', () => {
    it('should be defined', () => {
        expect(PagesMenu).toBeDefined();
    });

    it('it should render a dropdown align menu', () => {
        const tree = shallow(
        <PagesMenu
                alignMode="parent" />
        );
        expect(toJson(tree)).toMatchSnapshot();
    });
});