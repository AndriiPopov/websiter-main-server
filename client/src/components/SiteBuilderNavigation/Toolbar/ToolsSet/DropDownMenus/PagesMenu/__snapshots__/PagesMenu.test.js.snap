// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuilderZoomMenu it should render a dropdown align menu 1`] = `
<table>
  <tbody>
    <tr>
      <td>
        <Connect(BigButton)
          clicked="builderZoomOut"
          icon="<svg height=\\"30\\" viewBox=\\"0 0 24 24\\" width=\\"30\\" style=\\"transform: rotate(-90deg);\\"><path d=\\"M8 11h3v10h2V11h3l-4-4-4 4zM4 3v2h16V3H4z\\"></path></svg>"
        />
      </td>
      <td />
      <td>
        <Connect(BigButton)
          clicked="builderZoomIn"
          icon="<svg height=\\"30\\" viewBox=\\"0 0 24 24\\" width=\\"30\\" style=\\"transform: rotate(-90deg);\\"><path d=\\"M8 19h3v4h2v-4h3l-4-4-4 4zm8-14h-3V1h-2v4H8l4 4 4-4zM4 11v2h16v-2H4z\\"></path></svg>"
        />
      </td>
      <td>
        <Connect(BigButton)
          clicked="builderZoomReset"
          icon="<svg height=\\"30\\" viewBox=\\"0 0 24 24\\" width=\\"30\\" style=\\"transform: rotate(-90deg);\\"><path d=\\"M8 11h3v10h2V11h3l-4-4-4 4zM4 3v2h16V3H4z\\"></path></svg>"
        />
      </td>
    </tr>
  </tbody>
</table>
`;
